### PR TITLE
nixos/oomd: monitor some slices by default

### PR DESCRIPTION
**Pull request by m-bdf**
###### Description of changes

Following the new behaviour of Fedora 38, this makes systemd-oomd:
- monitor the system slice by default
- use the memory pressure policy instead of the swap policy
- make use of its new support for monitoring user-owned services

https://src.fedoraproject.org/rpms/systemd/c/7665e1796f915dedbf8e014f0a78f4f576d609bb 
https://src.fedoraproject.org/rpms/systemd/c/1320fc300948e7c12d16ea8dd4e0fae3fd821d54
https://src.fedoraproject.org/rpms/systemd/c/806c95e1c70af18f81d499b24cd7acfa4c36ffd6

For context, see discussion at https://github.com/NixOS/nixpkgs/pull/169613#issuecomment-1493826723

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
User slices are problematic and depend on a proper launcher and terminals that support launching everything in cgroups. Should likely only be a default, when on KDE and GNOME, but nowhere else.

cc @jtojnar @K900

**Comment by K900** [comment]
I can vouch for Plasma doing the right thing here, but I'm not sure if we should make this conditional on a desktop being _enabled_, and we can't really know if it's _actually running_.

**Comment by flokli** [comment]
How does it behave on Fedora for these DEs? Do they come with instructions to disable they by-default-being-on behaviour, or do they enable it only when KDE or GNOME are enabled?

If this is an unconditional default in Fedora, these DEs behaving problematic in OOM situations won't be a limited to NixOS problem, and these DEs will eventually support it. It's something they need to fix, and if we can't reliably do this automatically, I'd rather leave changing your system config to accommodate the chosen DE to the user, and mention it in the release notes.

**Comment by m-bdf** [comment]
Note that this change proposal was last updated on 2021-03-30, so it doesn't include the recent changes this PR brings to NixOS.

It's still informative about enabling oomd by default though, for instance [here](https://fedoraproject.org/wiki/Changes/EnableSystemdOomd#How_will_this_work_if_everything_is_in_the_same_cgroup?):
> #### How will this work if everything is in the same cgroup?
> It will not work as systemd-oomd acts on a per-cgroup level. Applications will need to spawn processes into separate cgroups (e.g. with systemd-run) or use a desktop environment (e.g. GNOME, KDE) that does this for them.

So should we make `enableUserSlices` opt-in by default?

**Comment by peterhoeg** [comment]
In terms of the opt-in vs opt-out, we need to cater for 3 user scenarios:

1. uses either GNOME or KDE exclusively
2. uses both one of GNOME/KDE and something else such as sway - chosen at the time of login.
3. uses !GNOME and !KDE exclusively

I would argue that we could check what environments have been enabled and only set `enableUserSlices` to true in that case, but it isn't really following the principle of least surprise.

I'd say make it opt-in.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "svg.preview.background": "transparent"
+}
```
Comment (flokli) [inline .vscode/settings.json:3]: Unrelated ;-)

Comment (m-bdf) [reply .vscode/settings.json]: Don't really know how this was committed, but anyway I just started all over again :)

**Comment by m-bdf** [comment]
Just noticed a [new commit](https://src.fedoraproject.org/rpms/systemd/c/806c95e1c70af18f81d499b24cd7acfa4c36ffd6) from a week ago that raises `ManagedOOMMemoryPressureLimit` from 50% to 80%.
I'll replicate this here, tell me if you think it's wrong.

**Comment by pearsche** [comment]
I was just looking into configuring systemd-oomd after Pitivi locked my laptop... Any news on this?

**Comment by m-bdf** [comment]
Implemented in #273921.
